src/tests/EdFi.DataManagementService.Tests.E2E/Features/References/UpdateReferenceValidation.feature

```diff
@@ -175,14 +172,16 @@ Feature: Update Reference Validation
                       "beginDate": "2021-08-23"
                   }
                   """
-             Then it should respond with 409
+             Then it should respond with 400
               And the response body is
                   """
                   {
-                      "detail": "The referenced 'SectionReferenceSectionReference' item does not exist.",
-                      "type": "urn:ed-fi:api:data-conflict:unresolved-reference",
-                      "title": "Unresolved Reference",
-                      "status": 409,
-                      "correlationId": null
+                      "detail":"Identifying values for the StudentSectionAssociation resource cannot be changed. Delete and recreate the resource item instead.",
+                      "type":"urn:ed-fi:api:bad-request:data-validation-failed:key-change-not-supported",
+                      "title":"Key Change Not Supported",
+                      "status": 400,
+                      "correlationId": null,
+                      "validationErrors":null,
+	                    "errors":null
```

Comment:
Interestingly, `StudentSectionAssociation` is also one of the few entities where we will need to allow cascading updates in the future. Today, we do get this new error message above - but the old one is what we should get in the future. So we'll need to `@ignore` this test until we work on cascading updates in DMS-80.

Please restore the old expected response, except that it had an error. The detail should be `The referenced 'Section' item does not exist.`.

`key-change-not-supported` is an important scenario, and it is covered in scenario 4.